### ISSUE TITLE
Custom Detekt task doesn't respect `include()` function

### ISSUE THREAD

**Issue by oheyadam** [detekt/detekt#7023]
## Expected Behavior
I have this custom detekt task to which I pass only the files that were changed because I don't want to run full `detekt`. I'm passing the files via [include()](https://detekt.dev/docs/gettingstarted/gradle/#leveraging-gradles-sourcetask---excluding-and-including-source-files). I expect it to run against this subset of files only

## Observed Behavior
Running the task with ``--info` says it has `NO-SOURCE`, even though I am passing a list of files.

## Steps to Reproduce
Create the task as such:
```kotlin
    val changedFilesProvider = providers.of(GitChangedFilesValueSource::class) {
        parameters.rootDir.set(rootProject.layout.projectDirectory.asFile.name)
    }
    tasks.register<Detekt>("fastDetekt") {
        changedFilesProvider.get()
            .forEach { file -> include(file) }
    }
```

`GitChangedFilesValueSource` is defined as such:

```kotlin
abstract class GitChangedFilesValueSource : ValueSource<List<String>, Parameters> {

    @get:Inject
    abstract val execOperations: ExecOperations

    interface Parameters : ValueSourceParameters {
        val rootDir: Property<String>
    }

    override fun obtain(): List<String> {
        val output = ByteArrayOutputStream()
        execOperations.exec {
            commandLine("git", "diff", "--name-only", "--diff-filter=ACM")
            standardOutput = output
        }
        val result = String(output.toByteArray(), Charset.defaultCharset())
            .trim()
            .split("\n")
            .map { line -> "${parameters.rootDir.get()}/$line" }
        println("Changed files: $result") // this is here for debugging. it prints the correct list of files
        return result
    }
}
```

Run `./gradlew fastDetekt`, it always succeeds even if you intentionally break some detekt rules in any of your files

## Your Environment

* Version of detekt used: 1.23.5
* Version of Gradle used (if applicable): 8.5
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: MacOS Sonoma 14.3.1
* Link to your project (if it's a public repository):

**Comment by 3flex** [comment]
Includes/excludes applies filters to the paths passed to `source` - you should pass the files you wish to scan to that parameter.

**Comment by oheyadam** [comment]
> Includes/excludes applies filters to the paths passed to `source` - you should pass the files you wish to scan to that parameter.

Sorry I failed to mention that I have a global Detekt config block that passes the source. Also doesn't detekt have a default value for this anyway?

I already have the following:

```kotlin
tasks.withType<Detekt>.configureEach {
    setSource(project.layout.projectDirectory)
}
```

Typing down this code on my phone, so it might be wrong, but you get the idea. I even copied this over to my custom task and it still doesn't work.

**Comment by 3flex** [comment]
Can you please put a self contained reproducer together? That will make it much easier to have a look at what's going on.

**Comment by oheyadam** [comment]
Hi @3flex, sorry for the delay. [Here's](https://github.com/android/nowinandroid/compare/main...oheyadam:nowinandroid:detekt-repro) a reproducer. An easy way to trigger this behavior is to add [this](https://github.com/android/nowinandroid/compare/main...oheyadam:nowinandroid:detekt-repro#diff-705d6b4cbfd6b5724912a84fedddac4c41309ddfb4c5f0950e6e7bf1d6d4371aL59) line back to `MainActivity` and run `./gradlew detekt`. This fails as expected, but running `./gradlew fastDetekt` doesn't. Thanks

**Comment by oheyadam** [comment]
Hey @3flex, just wanted to add another thing I noticed. In the same project structure that I linked, the `detektBaseline` function doesn't seem to work. Every time you invoke it, it just generates an empty file, even if you have new issues in your project.

What's wrong here? Does Detekt just not play nice when used in included builds?
